addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1209,7 +1209,9 @@ var SnippetEditor = Widget.extend({
                     // If the column doesn't come from a grid mode snippet.
                     if (!self.$target[0].classList.contains('o_grid_item')) {
                         // Converting the column to grid.
+                        self.options.wysiwyg.odooEditor.observerActive('dragAndDropMoveSnippet');
```

Comment:
I think you might have briefly explained it to me but I think I might need deeper details. Why do we need all of these around grid code?

Reply:
Before I added the grid mode to the drag and drop code, it was like this: 
- Deactivate the observer at the start of the drag (observerUnactive) because we don't need to observe anything during the drag
- Reactivate it at the drop, when it is inserted in its new snippet (so only the change of siblings is recorded, which is enough to undo the simple drag and drop).

When I added the grid code, I needed to observe the style changes and all (like the grid-area changes) to undo a drag properly, but since the observer was unactive, it was not working. So I simply made the observer always active (to see all these changes).
But the problem with that is that everything was observed, event the insertion of the `<we-hook>` elements, where the dropzones will then be inserted. So this caused mutations in elements that were not "concerned" by the current drag and drop to be recorded too, which would create unnecessary views in the backend. (see PR [1] and [2] for more details :wink: )

Therefore, I made the observer unactive again and simply observed what needed to be to undo properly the grid-related changes, which is why there are a lot of observer (de)activations :slightly_smiling_face: 


[1]: https://github.com/odoo/odoo/pull/106029
[2]: https://github.com/odoo/odoo/pull/106190